awx/ui_next/src/screens/Template/shared/SurveyList.jsx

```diff
@@ -1,48 +1,259 @@
-import React, { useEffect, useCallback } from 'react';
+import React, { useEffect, useCallback, useState } from 'react';
 import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
 
-import useRequest from '@util/useRequest';
+import useRequest, { useDeleteItems } from '@util/useRequest';
+import { Switch, Checkbox, Button } from '@patternfly/react-core';
+import {
+  DataToolbar,
+  DataToolbarContent,
+  DataToolbarGroup,
+  DataToolbarItem,
+} from '@patternfly/react-core/dist/umd/experimental';
 import ContentError from '@components/ContentError';
 import ContentLoading from '@components/ContentLoading';
+import ErrorDetail from '@components/ErrorDetail';
 import { JobTemplatesAPI } from '@api';
+import ContentEmpty from '@components/ContentEmpty';
+import { ToolbarAddButton } from '@components/PaginatedDataList';
+import { getQSConfig } from '@util/qs';
 import SurveyListItem from './SurveyListItem';
 
-function SurveyList({ template }) {
+import AlertModal from '../../../components/AlertModal';
+
+const QS_CONFIG = getQSConfig('survey', {
+  page: 1,
+});
+
+function SurveyList({ template, i18n }) {
+  const [selected, setSelected] = useState([]);
+  const [surveyEnabled, setSurveyEnabled] = useState(template.survey_enabled);
+  const [showToggleError, setShowToggleError] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+
   const {
-    result: questions,
+    result: { questions, name, description },
     error: contentError,
     isLoading,
     request: fetchSurvey,
   } = useRequest(
     useCallback(async () => {
       const {
-        data: { spec },
+        data: { spec, description: surveyDescription, name: surveyName },
       } = await JobTemplatesAPI.readSurvey(template.id);
-
-      return spec.map((s, index) => ({ ...s, id: index }));
-    }, [template.id])
+      return {
+        questions: spec?.map((s, index) => ({ ...s, id: index })),
```

Comment:
The use of  optional chaining (`spec?.map` as opposed to `spec.map` leads me to believe that you are anticipating scenarios where `spec` might not be defined.

If that is the case, you'd be passing something that  _isn't an array_ as the value as the value for `questions`. This is not good, because [much of the code that uses the questions variable](https://github.com/ansible/awx/blob/4d334991d2faf8f0686558c64e2cddfce5b3785a/awx/ui_next/src/screens/Template/shared/SurveyList.jsx#L71) expects it to be an array and will break otherwise.

Reply:
Yes there are cases when there are no questions, but a survey exists. This is good insight. Thanks